From d101ec8dbab7a6a9a817a64cd142b8f23618c1de Mon Sep 17 00:00:00 2001
From: Jayachandran B <jayachandran.b@intel.com>
Date: Tue, 14 Oct 2014 18:44:56 +0530
Subject: [PATCH 7/7] hdmi-audio: Query for HDMI pipe register base address

Issue: GMINL-4194

Currently, HDMI audio register base is hardcoded in hdmi audio driver
However there is a need to get the register base info from display
driver because:
a.	The base address may vary depending on the platform
b.	Register base depends on the display pipe used.
	The display driver may choose to use different pipes across
	HW variants of same platform or based on current display
	port assignment.
In this patch, hdmi audio driver queries display driver to get
the appropriate base address instead of hardcoding it.

Change-Id: I09fb16bb70e7c6c5dc7c40727cb8fb8a0342c957
Signed-off-by: Jayachandran B <jayachandran.b@intel.com>
Signed-off-by: meenakshiX, mali <meenakshix.mali@intel.com>
---
 sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c |   10 ++++++++--
 sound/drivers/intel_hdmi/intel_mid_hdmi_audio.h |    2 +-
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c b/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c
index f756b6e..b7e5f31 100644
--- a/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c
+++ b/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c
@@ -1840,8 +1840,14 @@ static int hdmi_audio_probe(struct platform_device *devptr)
 		goto err;
 	}
 	intelhaddata->hw_silence = 1;
-	/* PIPE B is used for HDMI*/
-	intelhaddata->audio_reg_base = 0x65800;
+
+	/* Query display driver for audio register base */
+	if (intelhaddata->reg_ops.hdmi_audio_get_register_base
+			(&intelhaddata->audio_reg_base)) {
+		pr_err("Unable to get audio reg base from Display driver\n");
+		goto err;
+	}
+
 	intelhaddata->ops = &had_ops_v2;
 	return retval;
 
diff --git a/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.h b/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.h
index a1a9383..ba38dca 100644
--- a/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.h
+++ b/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.h
@@ -553,7 +553,7 @@ struct snd_intelhad {
 	struct device *dev;
 	struct snd_kcontrol *kctl;
 	struct snd_pcm_chmap *chmap;
-	unsigned int	audio_reg_base;
+	uint32_t	audio_reg_base;
 	bool		hw_silence;
 	struct had_ops	*ops;
 };
-- 
1.7.9.5

